src/domains/queries.ml

```diff
@@ -42,13 +42,13 @@ module Unit = Lattice.Unit
     - whenever g is accessed and there are any threads other than the main thread that are created but not joined yet, m is held
 *)
 module Protection = struct
-  type t = Strong | Weak [@@deriving ord, hash]
+  type protection = Strong | Weak [@@deriving ord, hash]
```

Comment:
Why rename it away from `Protection.t`, which is the standard pattern? `Protection.protection` has such redundancy.
If it's just to avoid some `open`ing issues, then that's a clear sign to not use `open` so much.

Reply:
It's not avoiding issues per se, but writing `Protection.*` everywhere is annoying, so I want to open them. But then having a type flying around in a module that confusingly comes from some other module is just annoying.

Reply:
I renamed it back because due to type-directed disambiguation there's rarely the need to write out `Queries.Protection.Strong`.